uint32 peer_id
string task_guid